scalyr_agent/builtin_monitors/syslog_monitor.py

```diff
@@ -1095,6 +1095,7 @@ def worker(queue, is_shutdown):
                             global_log.info("ThreadPool shutting down, skipping further request processing.")
```

Comment:
I assume this means any data which is still in queue will be lost, right?

I think that's probably fine, but I wonder if it would make sense to update the shutdown logic to work something like that (if that's not already the case):

1. Stop syslog server which is accepting data (so producers / lb in front / similar can route data to another instance or similar (only work in case of TCP receiver of course)
2. Try to process what we have in queue for up to GRACE_PERIOD seconds
3. After grace period has been reached, exit regardless

IIRC, we already have a similar graceful shutdown process in place with the agent itself, but I don't remember how exactly it works and interacts with the monitors manager itself.

Maybe I'm also over complicating it a bit since likely a lot of producers only support UDP where that doesn't matter anyway...

Reply:
Monitor manager calls the stop method of each monitor and then, if configured so, waits for a monitors to stop.

I agree  it's a good idea to let the syslog monitor process the requests already received to avoid loosing any data. I will add it there. It's a good point with the LB, I can already see the Support Ticket describing a strange behavior, lost data and us spending 1 day figuring out, what happened :)